src/logcollector/lccom.c

```diff
@@ -33,13 +34,32 @@ size_t lccom_dispatch(char * command, char ** output){
         }
         return lccom_getconfig(rcv_args, output);
 
+    } else if (strcmp(rcv_comm, "getstate") == 0) {
+        return lccom_getstate(output);
     } else {
         mdebug1("LCCOM Unrecognized command '%s'.", rcv_comm);
         os_strdup("err Unrecognized command", *output);
         return strlen(*output);
     }
 }
 
+size_t lccom_getstate(char ** output) {
+    cJSON * state_json = NULL;
+    cJSON * w_packet = cJSON_CreateObject();
+    if (state_json = w_logcollector_state_get(), state_json == NULL) {
+        cJSON_AddNumberToObject(w_packet, "error", 1);
```

Comment:
I would replace **errors ID** with "self documented" **_defines_** for better readability